app_flutter/lib/logic/qualified_task.dart

```diff
@@ -38,37 +35,23 @@ class QualifiedTask {
 
   /// Get the URL for the configuration of this task.
   ///
-  /// Devicelab tasks are stored in the flutter/flutter Github repository.
   /// Luci tasks are stored on Luci.
   /// Cirrus tasks are stored on Cirrus.
-  ///
-  /// Throws [Exception] if [stage] does not match any of the above sources.
   String get sourceConfigurationUrl {
-    if (isLuci || isCirrus) {
-      return _externalSourceConfigurationUrl;
-    }
-    return '$_flutterGithubSourceUrl/dev/devicelab/bin/tasks/$task.dart';
-  }
-
-  String get _externalSourceConfigurationUrl {
     assert(isLuci || isCirrus);
-    switch (stage) {
-      case StageName.cirrus:
-        return '$_cirrusUrl/master';
-      case StageName.luci:
-        return '$_luciUrl/builders/luci.flutter.prod/$builder';
+    if (isCirrus) {
+      return '$_cirrusUrl/master';
+    } else if (isLuci) {
+      return '$_luciUrl/builders/luci.flutter.prod/$builder';
     }
     throw Exception('Failed to get source configuration url for $stage.');
   }
 
-  /// Whether this task is run in the devicelab or not.
-  bool get isDevicelab => stage.contains(StageName.devicelab);
-
   /// Whether this task was run on Cirrus CI.
   bool get isCirrus => stage == StageName.cirrus;
 
   /// Whether the task was run on the LUCI infrastructre.
-  bool get isLuci => stage == StageName.luci;
+  bool get isLuci => stage == StageName.cocoon || stage == StageName.legacyLuci || stage == StageName.luci;
```

Comment:
This is really the only change, everything else is deleting devicelab and defaulting to LUCI for certain logic.